src/prof_data.c

```diff
@@ -1167,9 +1167,11 @@ prof_tdata_init_impl(tsd_t *tsd, uint64_t thr_uid, uint64_t thr_discrim,
 	cassert(config_prof);
 
 	/* Initialize an empty cache for this thread. */
-	tdata = (prof_tdata_t *)iallocztm(tsd_tsdn(tsd), sizeof(prof_tdata_t),
-	    sz_size2index(sizeof(prof_tdata_t)), false, NULL, true,
-	    arena_get(TSDN_NULL, 0, true), true);
+	tdata = (prof_tdata_t *)iallocztm(tsd_tsdn(tsd),
+	    sizeof(prof_tdata_t) + sizeof(void *) * opt_prof_bt_max,
```

Comment:
nit: `sizeof(prof_tdata_t) + sizeof(void *) * opt_prof_bt_max` shows up twice. Make a variable for better readability.